src/pybind/mgr/volumes/fs/operations/versions/subvolume_v2.py

```diff
@@ -370,13 +341,17 @@ def remove(self, retainsnaps=False, internal_cleanup=False):
                 # Delete the volume meta file, if it's not already deleted
                 self.auth_mdata_mgr.delete_subvolume_metadata_file(self.group.groupname, self.subvolname)
                 return
+
         if self.state != SubvolumeStates.STATE_RETAINED:
-            self.trash_incarnation_dir()
-            self.metadata_mgr.update_global_section(MetadataManager.GLOBAL_META_KEY_PATH, "")
-            self.metadata_mgr.update_global_section(MetadataManager.GLOBAL_META_KEY_STATE, SubvolumeStates.STATE_RETAINED.value)
-            self.metadata_mgr.flush()
-            # Delete the volume meta file, if it's not already deleted
-            self.auth_mdata_mgr.delete_subvolume_metadata_file(self.group.groupname, self.subvolname)
+            self.trash_incarnation_dir(retainsnaps)
+            # Update metadata only for retainsnaps, don't update metadata on suvolume
```

Comment:
I'm unable to follow this comment. 
"don't update metadata on suvolume removal with group quota set, it fails when the group quota is exceeded."

So if a subvolume group has exceeded its quota, then the subvolume within the group cannot be removed ?

Reply:
PTAL and mark it resolved, if the rephrase looks fine.

Reply:
Good catch! I will test this out. I think this would fail. The correct design is to keep the internal metadata files out of the quota radar but unfortunately `.meta` file resides inside subvolume directory. I will think over it and update what can be done.

Reply:
Updating discussion happened on IRC here.

Yes, this would fail. If the group quota is exceeded, removing a subvolume with retainsnapshot fails. Any subvolume operation which updates the metadata would fail. 

The following options were considered to resolve this
1. Pre-allocate the subvolume metadata file:
             Didn't work. Python configparser failed to parse the pre-allocated config file. The config file is preallocated with default args, where the inode size is updated and bytes are filled with zeros. Updating inode size is required to avoid enforcing quota.
             Also, even if it works, pre-allocation needs to be done for all the existing subvolumes while setting quota on subvolumegroup which might not scale if there are thousands of subvolumes.
             
 2. Don't enforce quota if it's mgr client:
              The quota is enforced on the client and ceph doesn't distinguish between internal and external clients.
 
 3. Re-use/Over load existing FULL caps:
              The FULL caps, currently allow the write even if the osds are full if the client holds it. We can overload the same to not enforce quota as well. @gregsfortytwo thinks it's messy?

Reply:
- option 1 would surely give headaches (even if we can make it work)
- option 2 is probably alright, but Option 3 seems like the cleanest of the "hacks'.

Reply:
I was exploring further on this, thought of adding client option to disable quota. While doing so, stumbled on the commit https://github.com/ceph/ceph/pull/14978/commits/da9f1d829484fe35e6d5839c19afa371e444c194 which made the quota non-optional. If we can make quota optional again, and disable it for`mgr` client, a lot of issues which pop up because of setting quota on subvolumegroup goes away as discussed below.

1. We no longer need the commit https://github.com/ceph/ceph/commit/420541369cb7c09becc4c6fde7d7d2f1b76e60f4
    which forces to keep the subvolume trash directory within subvolumegroup.
2. Because of 1, subvolumegroup deletion return EAGAIN until the `trash` directory is empty.

@vshankar @batrick @ajarr @gregsfortytwo What do you guys think ?

Reply:
If this special option can be _only_ for the libcephfs instance in ceph-mgr, that would be neat.

Maybe introduce a ceph user for the libcephfs instance in ceph-mgr?

Reply:
Updated the PR with the option to disable quota bytes enforcement. The option is disabled for mgr libcephfs connection.

Observations:
1. With this PR, the quota enforcement is disabled for all mgr libcephfs connections but doesn't affect other clients by default.
2. Doing `ceph config set client client_quota_bytes false` would affect all the fuse mounts.

Reply:
@vshankar @ceph/cephfs Observations (contd..)
3. The client config set can be done by the users who has `write` caps on the `mon` i.e `mon allow rw`. The same is applicable to LibcephFS connections as well. But the LibcephFS connection can also be initialized via rados object as done in the `mgr` module. In `mgr` module the `rados` object is initialized via a specially constructed context (https://docs.ceph.com/en/latest/mgr/modules/#accessing-rados-or-cephfs). I believe it's applicable even for the LibCephFS connection which is initailized via rados object ? 
4. A user with `mon allow r` caps can mount the CephFS volume. So I believe giving out `mon allow rw` is giving out part of `admin` access which is not done unless required.

So I believe we are good to make quota optional and only the user with `mon allow rw` can enable/disable quota?